enterprise/kernel/src/test/java/org/neo4j/kernel/impl/enterprise/lock/forseti/ForsetiFalseDeadlockTest.java

```diff
@@ -0,0 +1,388 @@
+                },
+        FORSETI
+                {
+                    @Override
+                    public Locks create( ResourceType resourceType )
+                    {
+                        return new ForsetiLockManager( resourceType );
+                    }
+                };
+
+        public abstract Locks create( ResourceType resourceType );
+    }
+
+    @Parameterized.Parameters( name = "{0}" )
+    public static Iterable<Object[]> parameters()
+    {
+        List<Fixture> fixtures = new ArrayList<>();
+
+        // During development I also had iteration counts 1 and 2 here, but they never found anything, so for actually
+        // running this test, I leave only iteration count 100 enabled.
+        int[] iterations = new int[]{100};
+        LockManager[] lockManagers = LockManager.values();
+        LockWaitStrategies[] lockWaitStrategies = LockWaitStrategies.values();
+        LockType[] lockTypes = LockType.values();
+        for ( LockManager lockManager : lockManagers )
+        {
+            for ( LockWaitStrategies waitStrategy : lockWaitStrategies )
+            {
+                for ( LockType lockTypeAX : lockTypes )
+                {
+                    for ( LockType lockTypeAY : lockTypes )
+                    {
+                        for ( LockType lockTypeBX : lockTypes )
+                        {
+                            for ( LockType lockTypeBY : lockTypes )
+                            {
+                                for ( int iteration : iterations )
+                                {
+                                    fixtures.add( new Fixture(
+                                            iteration, lockManager, waitStrategy,
+                                            lockTypeAX, lockTypeAY, lockTypeBX, lockTypeBY ) );
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return fixtures.stream().map( f -> new Object[]{f} ).collect( Collectors.toList() );
+    }
+
+    private static ExecutorService executor = Executors.newCachedThreadPool( r ->
+    {
+        Thread thread = new Thread( r );
+        thread.setDaemon( true );
+        return thread;
+    } );
+
+    public ForsetiFalseDeadlockTest( Fixture fixture )
+    {
+        this.fixture = fixture;
+    }
+
+    private final Fixture fixture;
+
+    /**
+     * This takes a fair bit of time, and we don't want to wait that long. But more importantly, false deadlocks are
+     * still only very unlikely; they are not impossible, though.
+     *
+     * So this test is technically flaky, but the probability is, I think, quite low for the 'mild' test, but it is too
+     * high for this aggressive test. So therefor I have marked it as @Ignored, but I still keep it here for the sake
+     * of this comment, and to allow others to run it and get a feel for the probabilities involved.
+     */
+    @Ignore
```

Comment:
But that's fine right for separate stress test build, as one we already have for backup, index population etc.